lib/sycamore/sycamore/transforms/extract_table.py

```diff
@@ -107,40 +112,24 @@ def bbox_to_coord(bbox):
                 element.text_representation = element.text_representation + footer.text + "\n"
 
             all_tables.append(element)
-
         return all_tables
 
     def extract_tables(self, document: Document) -> Document:
-        tables = self._extract(document)
+        textract_result = self.get_textract_result(document)
+        tables = self.get_tables_from_textract_result(textract_result)
         document.elements = document.elements + tables
         return document
 
 
-class CachedTextractTableExtractor(TableExtractor):
+class CachedTextractTableExtractor(TextractTableExtractor):
     """
-    TextractTableExtractor with S3 based cache support
-
-    For each document, CachedTextractTableExtractor follows a series of steps for table extraction
-     1. if cache exists for current document, get from cache and return, otherwise, go to step 2
-     2. if run_full_textract is enabled, call textractor on the whole document and go to step 5; otherwise, go to step 3
-     3. if any element in the document is not marked table, return, otherwise, go to step 4.
-     4. clip pages which contain tables and run table extraction using textractor
-     5. update cache accordingly based on textractor result, return updated document
-
-    Cached table is in Json format as below:
-    {
-      "tables": [
-        {
-          "page_number": xx,
-          "title": "xxxx",
-          "content": "xxx"
-          "bbox": [xxx, xxx, xxx, xxx],
-          "footers": ["xxx", "xxx"]
-        },
-        ...
-      ]
-    }
-    bbox format is in [left/width, top/height, right/width, bottom/height]
+    Extends TextractTableExtractor with S3 based cache support for raw Textract results.
+
+    CachedTextractTableExtractor overrides the 'get_textract_result' method by doing the following:
+     1. if cache hit for current document, get from cache and return, otherwise continue
+     2. if run_full_textract is enabled, call textractor on the whole document and go to step 4
+     3. else clip pages which contain tables and run table extraction using textractor
```

Comment:
is this intentional, I see we still do the clip pages

Reply:
Yeah that's point 3?